src/botPage/view/TradeInfoPanel/TradeTable.js

```diff
@@ -103,17 +105,40 @@ export default class TradeTable extends Component {
             }
         });
         globalObserver.register('contract.settled', contract => {
-            this.registerTimeout(api, contract);
+            const contractID = contract.contract_id;
+            this.settleContract(api, contractID);
         });
     }
-    registerTimeout = (api, contract) => {
-        setTimeout(() => {
-            const contractID = contract.contract_id;
-            this.refreshContract(api, contractID);
-        }, 3000);
-    };
+
+    async settleContract(api, contractID) {
+        let settled = false;
+        let delay = 3000;
+
+        const sleep = d => new Promise(resolve => setTimeout(() => resolve(), d));
```

Comment:
You can remove the `d` argument as the value can be read from the outer scope.

```suggestion
        const sleep = () => new Promise(resolve => setTimeout(() => resolve(), delay);
```